onecodex/models/collection.py

```diff
@@ -339,6 +339,9 @@ def _collate_results(self, metric=None, include_host=None):
 
         data = np.zeros((len(self._classifications), len(tax_info.index)), dtype=metric_dtype)
 
+        if metric in [Metric.AbundanceWChildren, Metric.Abundance]:
+            data[:] = np.nan
```

Comment:
`ndarray.fill()` is a bit faster ([link](https://stackoverflow.com/a/1704853/3776794)):

```python
data.fill(np.nan)
```

The only reason I mention it is because this part of the code is the slowest part when plotting lots of data